### PR TITLE
Fix logger.sh

### PR DESCRIPTION
**Pull request by rk-fsdev**

### CODE REVIEW

File: appenders/bash/logger.sh [hunk 1]
```diff
@@ -4,49 +4,60 @@
 # Export the variables: PULSE_COLLECTOR_HOST and PULSE_COLLECTOR_PORT for the host and port caontaing the Pulse
 # Log Collector
 # ```
-# export PULSE_COLLECTOR_HOST=host1.com
-# export PULSE_COLLECTOR_PORT=9004
+# export PULSE_LOG_COLLECTOR_HOST=edge2.valhalla.phdata.io
```
Comment (afoerster) [inline appenders/bash/logger.sh]: Remove 'edge2...' and replace with something generic

File: appenders/bash/logger.sh [hunk 2]
```diff
@@ -4,49 +4,60 @@
...
+# export PULSE_LOG_COLLECTOR_PORT=9004
+# export APPLICATION_NAME=pulse_test_default
 # ```
-# Then source this scrip in your own bash scrip
-# ``` 
-# source logger.sh
-# logger --category=category --level=level --message=message --threadname=threadname --application=application-name
```
Comment (afoerster) [inline appenders/bash/logger.sh]: Can you replace this with the new usage documentation?

File: appenders/bash/logger.sh [hunk 3]
```diff
@@ -4,49 +4,60 @@
...
-# ```
-set -euo pipefail
+
+declare ERROR_MESSAGE="Error! Message info is only possible for its argument."
 
 logger(){
-  if [ "$#" -lt 1 ]
+  if [ -z "$APPLICATION_NAME" ] || [ -z "$PULSE_LOG_COLLECTOR_HOST" ] || [ -z "$PULSE_LOG_COLLECTOR_PORT" ]
```
Comment (afoerster) [inline appenders/bash/logger.sh]: Nice.

File: appenders/bash/logger.sh [hunk 4]
```diff
@@ -4,49 +4,60 @@
...
-    shift
-  done
-
-  timestamp=$( date -u +"%Y-%m-%dT%H:%M:%SZ")
+pulse_debug(){
+  if [ "$#" -lt 1 ]
+  then
+    echo $ERROR_MESSAGE
+  else
+    echo "Pulse is debuging \"$1\" message"
```
Comment (afoerster) [inline appenders/bash/logger.sh]: These messages can be enabled by a `PULSE_DEBUG` variable but they shouldn't print all the time, users won't want to see them

Comment (afoerster) [inline appenders/bash/logger.sh]: nit: misspelling

File: appenders/bash/logger.sh [hunk 5]
```diff
@@ -4,49 +4,60 @@
...
+    echo $(logger "debug" "$1")
```
Comment (afoerster) [inline appenders/bash/logger.sh]: This also shouldn't echo out, unless something like PULSE_DEBUG is set

File: appenders/bash/logger.sh [hunk 6]
```diff
@@ -4,49 +4,60 @@
...
+  fi
+}
 
-  curl -s -X POST -H 'Content-Type: application/json' -d '{"category": "'$category'","timestamp": "'$timestamp'", "level": "'$level'", "message": "'$message'", "threadName": "'$threadName'"}' http://$PULSE_COLLECTOR_HOST:$PULSE_COLLECTOR_PORT/v2/event/$application
+pulse_warn(){
+  if [ "$#" -lt 1 ]
+  then
```
Comment (afoerster) [inline appenders/bash/logger.sh]: This code is repeated can you add it to the logger function?

Comment (rk-fsdev) [reply appenders/bash/logger.sh]: I will try.

File: appenders/bash/logger.sh [hunk 7]
```diff
@@ -1,52 +1,59 @@
...
+  # $2 = the number of funtion parameters
+  # $3 = Message to be logged
+  # $4 = Pulse Logging text
+  if [ -z "$APPLICATION_NAME" ] || [ -z "$PULSE_LOG_COLLECTOR_HOST" ] || [ -z "$PULSE_LOG_COLLECTOR_PORT" ]
   then
-  echo "Usage Info: logger --category=category --level=level --message=message --threadname=threadname --application=application-name"
-  exit 1
+    echo "Error! Env variable should have PULSE_LOG_COLLECTOR_HOST, PULSE_LOG_COLLETOR_PORT, and APPLICATION_NAME"
+  else
+    if [ "$2" -ne 1 ]
```
Comment (afoerster) [inline appenders/bash/logger.sh]: Why is this comparing to `1`?

File: appenders/bash/logger.sh [hunk 8]
```diff
@@ -1,52 +1,59 @@
...
+# pulse_info "info message"
+# pulse_warn "warn message"
+# pulse_trace "trace message"
+# ```
 # Export the variables: PULSE_COLLECTOR_HOST and PULSE_COLLECTOR_PORT for the host and port caontaing the Pulse
 # Log Collector
 # ```
-# export PULSE_COLLECTOR_HOST=host1.com
-# export PULSE_COLLECTOR_PORT=9004
+# export PULSE_LOG_COLLECTOR_HOST=valhalla.phdata.io
```
Comment (afoerster) [inline appenders/bash/logger.sh]: nit: use `host.com` instead

File: appenders/bash/logger.sh [hunk 9]
```diff
@@ -1,52 +1,59 @@
...
-  done
-
-  timestamp=$( date -u +"%Y-%m-%dT%H:%M:%SZ")
+pulse_debug(){
+  logger "debug" "$#" "$1" "debugging"
+}
 
-  curl -s -X POST -H 'Content-Type: application/json' -d '{"category": "'$category'","timestamp": "'$timestamp'", "level": "'$level'", "message": "'$message'", "threadName": "'$threadName'"}' http://$PULSE_COLLECTOR_HOST:$PULSE_COLLECTOR_PORT/v2/event/$application
+pulse_warn(){
+  logger "warn" "$#" "$1" "warning"
```
Comment (afoerster) [inline appenders/bash/logger.sh]: why is `$#` needed?

File: appenders/bash/logger.sh (hunk 7 above)
Comment (rk-fsdev) [reply appenders/bash/logger.sh]: @afoerster , `$2` is the number of `pulse_debug` function arguments. `pulse_debug` function can only have one argument, the message to be logged.

File: appenders/bash/logger.sh (hunk 9 above)
Comment (rk-fsdev) [reply appenders/bash/logger.sh]: @afoerster , `$#` is the number of function arguments, and `pulse_***` functions should have only one argument, the message to be logged, as I explained above.

**Pull request merged**